File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/util/AesUtils.java
package org.javamaster.invocationlab.admin.util;

import org.javamaster.invocationlab.admin.enums.CipherTypeEnum;
import lombok.SneakyThrows;
import org.apache.commons.codec.digest.DigestUtils;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import javax.xml.bind.DatatypeConverter;
import java.nio.charset.StandardCharsets;
import java.util.Arrays;

public class AesUtils {

    @SneakyThrows
    public static String encrypt(final String plainValue, CipherTypeEnum cipherType, String cipherKey) {
        if (null == plainValue) {
            return null;
        }
        try {
            byte[] result = getCipher(Cipher.ENCRYPT_MODE, cipherType, cipherKey)
                    .doFinal(plainValue.getBytes(StandardCharsets.UTF_8));
            return DatatypeConverter.printBase64Binary(result);
        } catch (Exception e) {
            return plainValue;
        }
    }

    @SneakyThrows
    public static String decrypt(final String cipherValue, CipherTypeEnum cipherType, String cipherKey) {
        if (null == cipherValue) {
            return null;
        }
        try {
            byte[] result = getCipher(Cipher.DECRYPT_MODE, cipherType, cipherKey)
                    .doFinal(DatatypeConverter.parseBase64Binary(cipherValue));
            return new String(result, StandardCharsets.UTF_8);
        } catch (Exception e) {
            return cipherValue;
        }
    }

    @SneakyThrows
    private static Cipher getCipher(final int decryptMode, CipherTypeEnum cipherType, String cipherKey) {
        byte[] secretKey;
        switch (cipherType) {
            case MYSQL_AES:
                secretKey = Arrays.copyOf(cipherKey.getBytes(StandardCharsets.UTF_8), 16);
                break;
            case AES:
                secretKey = Arrays.copyOf(DigestUtils.sha1(cipherKey), 16);
                break;
            default:
                throw new IllegalArgumentException(cipherKey);
        }
        Cipher result = Cipher.getInstance(CipherTypeEnum.AES.type);
        result.init(decryptMode, new SecretKeySpec(secretKey, CipherTypeEnum.AES.type));
        return result;
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/util/AesUtils.kt
package org.javamaster.invocationlab.admin.util

import org.javamaster.invocationlab.admin.enums.CipherTypeEnum
import org.apache.commons.codec.digest.DigestUtils
import java.nio.charset.StandardCharsets
import java.util.*
import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec

object AesUtils {
    @JvmStatic

    fun encrypt(plainValue: String?, cipherType: CipherTypeEnum, cipherKey: String): String? {
        if (null == plainValue) {
            return null
        }
        try {
            val result = getCipher(Cipher.ENCRYPT_MODE, cipherType, cipherKey)
                .doFinal(plainValue.toByteArray(StandardCharsets.UTF_8))
            return Base64.getEncoder().encodeToString(result)
        } catch (e: Exception) {
            return plainValue
        }
    }

    @JvmStatic

    fun decrypt(cipherValue: String?, cipherType: CipherTypeEnum, cipherKey: String): String? {
        if (null == cipherValue) {
            return null
        }
        try {
            val result = getCipher(Cipher.DECRYPT_MODE, cipherType, cipherKey)
                .doFinal(Base64.getDecoder().decode(cipherValue))
            return String(result, StandardCharsets.UTF_8)
        } catch (e: Exception) {
            return cipherValue
        }
    }


    private fun getCipher(decryptMode: Int, cipherType: CipherTypeEnum, cipherKey: String): Cipher {
        val secretKey = when (cipherType) {
            CipherTypeEnum.MYSQL_AES -> cipherKey.toByteArray(StandardCharsets.UTF_8).copyOf(16)
            CipherTypeEnum.AES -> DigestUtils.sha1(cipherKey).copyOf(16)
        }
        val result = Cipher.getInstance(CipherTypeEnum.AES.type)
        result.init(decryptMode, SecretKeySpec(secretKey, CipherTypeEnum.AES.type))
        return result
    }
}
